diff --git a/Configurations/unix-Makefile.tmpl b/Configurations/unix-Makefile.tmpl
index e52ff981a..e9ba873c6 100644
--- a/Configurations/unix-Makefile.tmpl
+++ b/Configurations/unix-Makefile.tmpl
@@ -614,13 +614,6 @@ install_dev: install_runtime_libs
 		cp $$i "$(DESTDIR)$(INSTALLTOP)/include/openssl/$$fn"; \
 		chmod 644 "$(DESTDIR)$(INSTALLTOP)/include/openssl/$$fn"; \
 	done
-	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(INSTALLTOP)/include/oqs"
-	@set -e; for i in $(SRCDIR)/oqs/include/oqs/*.h; do \
-		fn=`basename $$i`; \
-		$(ECHO) "install $$i -> $(DESTDIR)$(INSTALLTOP)/include/oqs/$$fn"; \
-		cp $$i "$(DESTDIR)$(INSTALLTOP)/include/oqs/$$fn"; \
-		chmod 644 "$(DESTDIR)$(INSTALLTOP)/include/oqs/$$fn"; \
-	done
 	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(libdir)"
 	@set -e; for l in $(INSTALL_LIBS); do \
 		fn=`basename $$l`; \
@@ -780,25 +773,24 @@ install_runtime_libs: build_libs
 		: {- output_on() if windowsdll(); "" -}; \
 	done
 ifeq (,$(OQSLIBDIR))
-	@$(ECHO) "No OQS library directory found to install. Exiting install with failure."
-	exit 1
-endif
-ifneq (,$(wildcard $(OQSLIBDIR)/liboqs.a))
-	$(ECHO) "install $(OQSLIBDIR)/liboqs.a $(DESTDIR)$(libdir)"; \
-	install $(OQSLIBDIR)/liboqs.a $(DESTDIR)$(libdir)
-endif
-ifneq (,$(wildcard $(OQSLIBDIR)/liboqs.so.0*))
-	@set -e; for i in $(OQSLIBDIR)/liboqs.so.0.*; do \
-		fn=`basename $$i`; \
-		sn=`basename $(OQSLIBDIR)/liboqs.so.?`; \
-		$(ECHO) "install $(OQSLIBDIR)/$$fn $(DESTDIR)$(libdir)"; \
-		$(ECHO) "cd $(DESTDIR)$(libdir) && ln -sf $$fn liboqs.so && ln -sf $$fn $$sn && cd -"; \
-		install $(OQSLIBDIR)/$$fn $(DESTDIR)$(libdir); \
-		cd $(DESTDIR)$(libdir) && ln -sf $$fn liboqs.so && ln -sf $$fn $$sn && cd -; \
-	done
+	@$(ECHO) "No OQS library directory found to install."
+else
+	ifneq (,$(wildcard $(OQSLIBDIR)/liboqs.a))
+		$(ECHO) "install $(OQSLIBDIR)/liboqs.a $(DESTDIR)$(libdir)"; \
+		install $(OQSLIBDIR)/liboqs.a $(DESTDIR)$(libdir)
+	endif
+	ifneq (,$(wildcard $(OQSLIBDIR)/liboqs.so.0*))
+		@set -e; for i in $(OQSLIBDIR)/liboqs.so.0.*; do \
+			fn=`basename $$i`; \
+			sn=`basename $(OQSLIBDIR)/liboqs.so.?`; \
+			$(ECHO) "install $(OQSLIBDIR)/$$fn $(DESTDIR)$(libdir)"; \
+			$(ECHO) "cd $(DESTDIR)$(libdir) && ln -sf $$fn liboqs.so && ln -sf $$fn $$sn && cd -"; \
+			install $(OQSLIBDIR)/$$fn $(DESTDIR)$(libdir); \
+			cd $(DESTDIR)$(libdir) && ln -sf $$fn liboqs.so && ln -sf $$fn $$sn && cd -; \
+		done
+	endif
 endif

-
 install_programs: install_runtime_libs build_programs
 	@[ -n "$(INSTALLTOP)" ] || (echo INSTALLTOP should not be empty; exit 1)
 	@$(PERL) $(SRCDIR)/util/mkdir-p.pl "$(DESTDIR)$(INSTALLTOP)/bin"
